controllers/nodeconfig/kubelet_config.go

```diff
@@ -0,0 +1,74 @@
+package nodeconfig
+
+import (
+	"context"
+	"encoding/json"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+
+	"github.com/kubevirt/hyperconverged-cluster-operator/pkg/metrics"
+)
+
+type kubeletConfig struct {
+	logger    logr.Logger
+	apiReader client.Reader
+	client    rest.Interface
+
+	KubeletConfig struct {
+		NodeStatusMaxImages int `json:"nodeStatusMaxImages"`
+	} `json:"kubeletconfig"`
+}
+
+func newKubeletConfig(logger logr.Logger, apiReader client.Reader, client rest.Interface) *kubeletConfig {
+	return &kubeletConfig{
+		logger:    logger,
+		apiReader: apiReader,
+		client:    client,
+	}
+}
+
+func (kc *kubeletConfig) updateNodeImageMetrics(ctx context.Context) {
+	nodeList := corev1.NodeList{}
+	err := kc.apiReader.List(ctx, &nodeList)
+	if err != nil {
+		kc.logger.Error(err, "Failed to list nodes")
+		return
+	}
+
+	for _, node := range nodeList.Items {
+		err = metrics.HcoMetrics.SetHCOMetricNumberOfImages(node.Name, len(node.Status.Images))
+		if err != nil {
+			kc.logger.Error(err, "Failed to set number of images metric")
+			continue
+		}
+
+		kc.setNodeMaxImagesMetrics(ctx, node.Name)
```

Comment:
I think we should cache it somehow in memory for each node and then we should refresh it only when we detect that the node configuration changed (not really sure how).